pkg/api/diagnostic.go

```diff
@@ -0,0 +1,68 @@
+package api
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	nurl "net/url"
+	"strconv"
+	"strings"
+
+	"github.com/wakatime/wakatime-cli/pkg/diagnostic"
+	"github.com/wakatime/wakatime-cli/pkg/log"
+	"github.com/wakatime/wakatime-cli/pkg/version"
+)
+
+// SendDiagnostics sends diagnostics to the WakaTime api.
+func (c *Client) SendDiagnostics(plugin string, diagnostics ...diagnostic.Diagnostic) error {
+	url := c.baseURL + "/v1/plugins/errors"
+
+	log.Debugf("sending diagnostic data to api at %s", url)
+
+	data := nurl.Values{}
+	data.Set("platform", version.OS)
+	data.Set("architecture", version.Arch)
+	data.Set("editor", plugin)
+
+	for _, d := range diagnostics {
+		switch d.Type {
+		case diagnostic.TypeLogs:
+			data.Set("logs", d.Value)
+		case diagnostic.TypeStack:
+			data.Set("stacktrace", d.Value)
+		default:
+			return fmt.Errorf("Unknown diagnostic type %d", d.Type)
+		}
+	}
+
+	req, err := http.NewRequest(http.MethodPost, url, strings.NewReader(data.Encode()))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %s", err)
+	}
+
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
```

Comment:
We might set `application/json` here as discussed [here](https://wakatime.slack.com/archives/GUWD67UE4/p1622765826010600).